### ISSUE TITLE
VPC Flow Log filter expression doen't allow to use "psc" field  

### ISSUE THREAD

**Issue by andy65535** [hashicorp/terraform-provider-google#19239]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team) to expedite investigation and resolution of this issue.

### Terraform Version & Provider Version(s)

Terraform v1.5.5
on MacOS
+ provider registry.terraform.io/hashicorp/google v5.42.0
+ provider registry.terraform.io/hashicorp/google-beta v5.42.0

### Affected Resource(s)

google_compute_subnetwork

### Terraform Configuration

```tf
```

### Expected Behavior

VPC flow logs should be modified to log only flow with PSC as source or destination. 

```
module "vpc-test" {
  source  = "terraform-google-modules/network/google"
  version = "~> 9.1"

  project_id   = var.gcp_project_id
  network_name = "test-vpc-01"
  routing_mode = "GLOBAL"

  subnets = [
    {
      subnet_name           = "subnet-04"
      subnet_ip             = "10.1.1.0/24"
      subnet_region         = local.region
      subnet_private_access = "true"
      subnet_flow_logs          = "true"
      subnet_flow_logs_sampling = 0.75
      subnet_flow_logs_interval = "INTERVAL_1_MIN"
      subnet_flow_logs_metadata = "INCLUDE_ALL_METADATA"
+      subnet_flow_logs_filter = "has(psc.psc_endpoint.vpc.vpc_name)"
    }
  ]

  secondary_ranges = {
  }
}
```

### Actual Behavior

```
Terraform will perform the following actions:

  # module.vpc-test.module.subnets.google_compute_subnetwork.subnetwork["europe-west2/subnet-04"] will be updated in-place
  ~ resource "google_compute_subnetwork" "subnetwork" {
        id                         = "projects/XXXXXXXXX/regions/europe-west2/subnetworks/subnet-04"
        name                       = "subnet-04"
        # (12 unchanged attributes hidden)

      ~ log_config {
          ~ filter_expr          = "true" -> "has(psc.psc_endpoint.vpc.vpc_name)"
            # (4 unchanged attributes hidden)
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.


Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.vpc-test.module.subnets.google_compute_subnetwork.subnetwork["europe-west2/subnet-04"]: Modifying... [id=projects/XXXXXXXXX/regions/europe-west2/subnetworks/subnet-04]
╷
│ Error: Error updating Subnetwork "projects/XXXXXXXXX/regions/europe-west2/subnetworks/subnet-04": googleapi: Error 400: Invalid value for field 'resource.logConfig.filterExpr': 'has(psc.psc_endpoint.vpc.vpc_name)'. ERROR: VPC Flow Log filter expression:1:5: undeclared reference to 'psc' (in container '')
│   | has(psc.psc_endpoint.vpc.vpc_name)
│   | ....^, invalid
│
│   with module.vpc-test.module.subnets.google_compute_subnetwork.subnetwork["europe-west2/subnet-04"],
│   on .terraform/modules/vpc-test/modules/subnets/main.tf line 28, in resource "google_compute_subnetwork" "subnetwork":
│   28: resource "google_compute_subnetwork" "subnetwork" {
│
```

If I use, f.e.
subnet_flow_logs_filter = "has(src_vpc.vpc_name)"
it works as expected.

### Steps to reproduce

1. `terraform apply`

### Important Factoids

I'm following the Google documentation page
https://cloud.google.com/vpc/docs/about-flow-logs-records#record_format
where "psc" described ad Metadata field and as far I have "INCLUDE_ALL_METADATA" I expected "psc" field will be there.
Also I see "psc" field in the Logs Explorer.

b/364879316

**Comment by ggtisc** [comment]
Hi @andy65535!

To replicate this issue we need the source code of your module `terraform-google-modules/network/google` and all other things that we don't have access to. If you have sensitive data you could use examples like:

project = "my-project"
org_id = 1234567890
user = "my-user@my-domain.com"

**Comment by andy65535** [comment]
> Hi @andy65535!
> 
> To replicate this issue we need the source code of your module `terraform-google-modules/network/google` and all other things that we don't have access to. If you have sensitive data you could use examples like:
>

It is not a my module. It is Google module: https://github.com/terraform-google-modules/terraform-google-network/tree/master
 
> project = "my-project" org_id = 1234567890 user = "[my-user@my-domain.com](mailto:my-user@my-domain.com)"

**Comment by ggtisc** [comment]
Thanks @andy65535, unfortunately we are not responsible for maintaining terraform modules. After searching I have not found the `google_compute_subnetwork` resource code configuration, so if you could provide the code you are using in that module, it will be helpful to reproduce this issue.

**Comment by andy65535** [comment]
Google fixed this in scope of Case 3154088.
This issue can be closed.

**Comment by andy65535** [comment]
Close.
